src/protocol/attribution/accumulate_credit.rs

```diff
@@ -261,11 +264,15 @@ mod tests {
     const H: [u128; 2] = [0, 1];
     const BD: [u128; 8] = [0, 1, 2, 3, 4, 5, 6, 7];
 
-    #[derive(Clone, Copy)]
-    struct Fp31AttributionInputRow([Fp31; 4]);
+    #[derive(Clone, Copy, Debug, PartialEq, Eq)]
```

Comment:
Can we lose the `Copy` ?

```suggestion
    #[derive(Clone, Debug, PartialEq, Eq)]
```